supervised/preprocessing/eda.py

```diff
@@ -162,3 +157,118 @@ def compute(X, y, eda_path):
+                        )
+
+                    elif PreprocessingUtils.get_type(X[col]) in (
+                        "categorical",
+                        "discrete",
+                    ):
+
+                        if np.nunique(X[col]) <= 7:
+
+                            plt.figure(figsize=(5, 5))
+                            sns.countplot(x=X[col], hue=y)
+                            plt.gca().set_title(
+                                f"Count plot of each {col}",
+                                fontsize=11,
+                                weight="bold",
+                                alpha=0.75,
+                            )
+
+            elif PreprocessingUtils.get_type(y) == "continous":
+
+                print("here")
+                for col in X.columns:
+
+                    if PreprocessingUtils.get_type(X[col]) == "continous":
+
+                        plt.figure(figsize=(5, 5))
+                        plt.scatter(X[col].values, y)
+                        plt.gca().set_xlabel(f"{col}")
+                        plt.gca().set_ylabel("target")
+                        plt.gca().set_title(
+                            f"Scatter plot of {col} vs target",
+                            fontsize=11,
+                            weight="bold",
+                            alpha=0.75,
+                        )
+
+                    elif PreprocessingUtils.get_type(X[col]) in (
+                        "categorical",
+                        "discrete",
+                    ):
+                        if X[col].nunique() <= 7:
+
+                            plt.figure(figsize=(5, 5))
+                            for i in X[col].unique():
+                                sns.kdeplot(
+                                    y[X[X[col] == i].index],
+                                    shade=True,
+                                    label=f"{col}_{i}",
+                                )
+                            plt.gca().set_title(
+                                f"Distribution of target for each {col}",
+                                fontsize=11,
+                                weight="bold",
+                                alpha=0.75,
+                            )
+                            plt.legend()
+
+                    elif PreprocessingUtils.get_type(X[col]) == "datetime":
+
+                        plt.figure(figsize=(5, 5))
+                        plt.plot(X[col], y)
+                        plt.gca().set_xticklabels(X[col].dt.date, rotation="45")
+                        plt.gca().set_title(
+                            f"Distribution of target over time",
+                            fontsize=11,
+                            weight="bold",
+                            alpha=0.75,
+                        )
+
+            cols = [
+                col
+                for col in X.columns
+                if PreprocessingUtils.get_type(X[col]) == "continous"
+            ][:14]
```

Comment:
why 14? how this number was calculated? You can assume that there is a machine with at least 4GB RAM available. It should go as constant (with capital letters) at the top of the file.